File: php/src/TwoPointer/ContainerWater.php
<?php

namespace IbrahimMH\LeetCode\Solution\TwoPointer;

class ContainerWater{

    public function calculateArea(array $list){

        $left = 0;
        $right = count($list) - 1;
        $maxArea = 0;

        while($left < $right){

            $area = min($list[$left],$list[$right]) *( $right - $left);
            $maxArea = max($maxArea,$area);

            if($list[$left] < $list[$right]){
                $left++;
            }else{
                $right--;
            }


        }
        return $maxArea;
    }
}

File: python/two_pointer/container_water.py



def calc_area(list)->int:

    left = 0
    right = len(list) - 1
    max_area = 0

    while left < right :

        area = min(list[left], list[right]) * (right - left)

        max_area = max(max_area, area)

        if list[left] < list[right]: 
            left += 1
        else:
            right -= 1

    return max_area